### PR TITLE
Features/api design

### PR DESCRIPTION
**Pull request by jaimin45**

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          schema:
+            $ref: '#/definitions/tutorial'
+definitions:
+  tutorial:
+    required:
+    - UUID
+    - createdAt
+    - updatedAt
+    properties:
+      UUID:
```
Comment (kapilbarad) [inline swagger.yaml]: The name of this field should be `_id` but the type should be `string` and format should be `uuid`.

```
_id: 
  type: string
  format: uuid
```

Same for id in other places.

File: swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        type: String
+        uniqueItems: true
+      isPublic:
```
Comment (kapilbarad) [inline swagger.yaml]: As per the schema, this should be:
```
published:
  type: boolean
```

File: swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+produces:
+- application/yaml
+paths:
+  tutorial:
+    parameters:
+    - name: id
+      in: path
+      required: true
+      description: ID of tutorial that we want to find
+      type: integer
```
Comment (kapilbarad) [inline swagger.yaml]: Should be:
```
type: string
format: uuid
```

File: swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  description: An app to manage Tutorials
+host: localhost:8000
+basePath: "/"
+tags:
+- name: tutorials
+  description: API for user in the system
+schemes:
+- http
+consumes:
+- application/yaml
```
Comment (kapilbarad) [inline swagger.yaml]: Should be `application/json`

File: swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+produces:
+- application/yaml
```
Comment (kapilbarad) [inline swagger.yaml]: Should be `application/json`

File: swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    - UUID
+    - createdAt
+    - updatedAt
+    properties:
+      UUID:
+        type: String
+        uniqueItems: true
+      isPublic:
+        type: boolean
+      created:
```
Comment (kapilbarad) [inline swagger.yaml]: Should be `createdAt`.

**Review by kapilbarad — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 4 above)
Comment (ghost) [reply swagger.yaml]: This is Request and response `Content-Type` which is in our case should be always json.

File: swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    - createdAt
+    - updatedAt
+    properties:
+      _id:
+        type: string
+        format: uuid
+        uniqueItems: true
+      published:
+        type: boolean
+      creatAt:
```
Comment (kapilbarad) [inline swagger.yaml]: Typo: createdAt

File: swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+      tags:
+      - tutorials
+      parameters:
+      - name: tutorial
+        in: body
+        description: tutorial with new values of properties
+        schema:
+          $ref: '#/definitions/tutorial'
+      responses:
+        '200':
```
Comment (kapilbarad) [inline swagger.yaml]: The Patch request should mostly return 200 without any response body. So no need to attach schema. for 200.

File: swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+          schema:
+            $ref: '#/definitions/tutorial'
+    delete:
+      summary: Delete tutorials with given ID
+      tags:
+      - tutorials
+      responses:
+        '200':
+          description: tutorial is deleted
+          schema:
```
Comment (kapilbarad) [inline swagger.yaml]: No need to attach schema here.

**Review by kapilbarad — CHANGES_REQUESTED** [review]

File: swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        '200':
+          description: tutorial is updated
+          schema:
+            $ref: '#/definitions/tutorial'
+definitions:
+  tutorial:
+    required:
+    - _id
+    - createdAt
+    - updatedAt
```
Comment (kapilbarad) [inline swagger.yaml]: `published` is also required.

**Review by kapilbarad — APPROVED** [review]

**Review by kapilbarad — APPROVED** [review]

**Comment by jaimin45** [comment]
docs: Added swagger yaml file

**Pull request merged**